install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -113,19 +113,10 @@ data:
   #   the kvstore by commenting out the identity-allocation-mode below, or
   #   setting it to "kvstore".
   identity-allocation-mode: {{ .Values.identityAllocationMode }}
-{{- if hasKey .Values "identityHeartbeatTimeout" }}
-  identity-heartbeat-timeout: "{{ .Values.identityHeartbeatTimeout }}"
-{{- end }}
-{{- if hasKey .Values "identityGCInterval" }}
-  identity-gc-interval: "{{ .Values.identityGCInterval }}"
-{{- end }}
-{{- if hasKey .Values.operator "endpointGCInterval" }}
-  cilium-endpoint-gc-interval: "{{ .Values.operator.endpointGCInterval }}"
-{{- end }}
-
-{{- if hasKey .Values.operator "nodeGCInterval" }}
-  nodes-gc-interval: "{{ .Values.operator.nodeGCInterval | default "0s" }}"
-{{- end }}
+  identity-heartbeat-timeout: {{ include "validateDuration" .Values.identityHeartbeatTimeout | quote }}
```

Comment:
I think we still need to check if the key exists or now. If the key is not available, the default value is coming from CLI code, but not 0s.

Reply:
There is one case can't be satisfied if we use the logic of if key not available, use default non zero value from CLI.
`hasDuration` check in `clusterrole.yaml` can lead to wrong result.
The assumption is when set value to null, it's equivalent to disable the feature, for duration field, it will use default 0 for that purpose.
So I feel if the value is exposed through helm, it should decouple from CLI, to have one source of truth.
This is to address the concern in https://github.com/cilium/cilium/pull/19576#pullrequestreview-955075988:
> Settting to null shows nothing, meaning we don't pass the flag and we don't have the RBAC permission. The issue here is that we'll be using the non-zero CLI default (5m) and will fail because we don't have the permission.

Reply:
@sayboras I just noticed `identity-heartbeat-timeout` and `identity-gc-interval` are not using the correct helm values, missing `operator` in the path, there are commented out root level values in 

https://github.com/cilium/cilium/blob/8e0197ad01cd168592399893e24e7fddfacb8b09/install/kubernetes/cilium/values.yaml#L1104-L1108

I updated the path when amending the code, not sure if this is expected.
If not, I can leave these 2 out of this PR as they are not documented fields.